testing/__init__.py

```diff
@@ -1,2 +1,14 @@
 """Utilities for tests in tests/*."""
 from __future__ import annotations
+
+import sys
+
+from testing.mocker_modules import pymargo_mocker
+from testing.mocker_modules import ucx_mocker
+
+# pymargo and ucx mocker imports
+
+sys.modules['pymargo'] = pymargo_mocker
```

Comment:
If I actually had all of these packages installed, the mocked ones would still get used here right?

Maybe we should make this a fixture in `testing/mocker_module/` that looks like this:
```python
@pytest.mark.fixture(autouse=True, scope='session'):
def mock_non_pip_modules() -> None:  # pragma: no cover
    """..."""
    try:
        import pymargo.bulk
    except ImportError:
        sys.modules['pymargo.bulk'] = pymargo_mocker

    ...  # rest of them
```
and then import this fixture in `tests/conftest.py`.

Reply:
I think I read this too quickly when I responded. You definitely bring up an excellent point, but the reason why I put it in the `__init__` and not elsewhere is because I don't want it to hit the `except` clause in `margo.py` when `store` is loaded. I struggled with modifying the module after `margo.py` is already loaded (would just tell me pymargo doesn't exist).

I'll try what you suggest just in case it was a bug/I had done something wrong and it does actually work.

Reply:
The `scope='session'` will ensure the fixture is run before any tests (and hopefully imports in any test files).